pkgs/tools/archivers/gnutar/default.nix

```diff
@@ -24,33 +30,49 @@ stdenv.mkDerivation rec {
   # gnutar tries to call into gettext between `fork` and `exec`,
   # which is not safe on darwin.
   # see http://article.gmane.org/gmane.os.macosx.fink.devel/21882
-  postPatch = lib.optionalString stdenv.isDarwin ''
+  postPatch = lib.optionals stdenv.isDarwin ''
     substituteInPlace src/system.c --replace '_(' 'N_('
   '';
 
   outputs = [ "out" "info" ];
 
-  nativeBuildInputs = lib.optional stdenv.isDarwin autoreconfHook;
+  nativeBuildInputs = [
+    makeBinaryWrapper
+  ] ++ lib.optionals stdenv.isDarwin [
+    autoreconfHook
+  ];
+
   # Add libintl on Darwin specifically as it fails to link (or skip)
   # NLS on it's own:
   #  "_libintl_textdomain", referenced from:
   #    _main in tar.o
   #  ld: symbol(s) not found for architecture x86_64
-  buildInputs = lib.optional stdenv.isLinux acl ++ lib.optional stdenv.isDarwin libintl;
+  buildInputs = lib.optionals stdenv.isLinux [
+    acl
+  ] ++ lib.optionals stdenv.isDarwin [
+    libintl
+  ];
 
   # May have some issues with root compilation because the bootstrap tool
   # cannot be used as a login shell for now.
   FORCE_UNSAFE_CONFIGURE = lib.optionalString (stdenv.hostPlatform.system == "armv7l-linux" || stdenv.isSunOS) "1";
 
-  preConfigure = if stdenv.isCygwin then ''
+  preConfigure = lib.optionals stdenv.isCygwin ''
```

Comment:
should be `lib.optionalString`

Reply:
I reverted it, there are also occurences of `preConfigure = lib.optionals ` in `nixpkgs`. We should fixed them as well.

Reply:
Feel free to go ahead in another PR. 👍